### PR TITLE
Add basic test for speech recognition and synthesis

### LINKED ISSUE #2122: Add to test framework: the ability to emulate Web Speech
## Feature Request
**Is your feature request related to a problem? Please describe.**
This issue is related to https://github.com/microsoft/BotFramework-WebChat/issues/2022 -- `inputHint` is not respected in speech scenarios. 2022 is to ensure that Web Chat does respect `inputHint`.

Because we are making changes to Speech, we want the ability to test the different scenarios that are possible with `inputHint` and activities from the bot that are spoken.

**Describe the suggestion or request in detail**

Please emulate speech activities that
-- allow checking of the queue
-- signal finishing of speech

**Describe alternatives you have considered**
While we will be able to do manual tests, we also want to add tests into our framework. 

[Enhancement]

(After these emulations are created, we will need to actually write those tests. At this point this issue will most likely be reassigned to me)

### PR DESCRIPTION
**Pull request by compulim**
Fixes #2122.

## Changelog Entry

-  Added test harness for speech recognition and synthesis for [#2122](https://github.com/microsoft/BotFramework-WebChat/issues/2122), by [@compulim](https://github.com/compulim), in PR [#2153](https://github.com/Microsoft/BotFramework-WebChat/pull/2153)

## Description

Speech recognition and synthesis in Web Chat is not tested due to lack of test harness.

This work will add test harness so we can start testing speech functionality in Web Chat, along with basic tests for speech.

This work does not modify any production code.

## Specific Changes

- Added `mockWebSpeech.js` while loading the test harness page
   - Exposed as `window.WebSpeechMock` for required ponyfills and mock functions
- Devs can ask the test harness for a couple of things
   - Recognize a single transcript
   - Recognize with unhappy paths (e.g. bird tweets or network error)
   - Get text that need to be synthesized
- Tests added
   - Basic test for multiple turn for speech recognition and synthesis
   - Test for interrupting speech recognition via keyboard

---

-  [x] Testing Added

### CODE REVIEW

File: __tests__/setup/web/mockWebSpeech.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      return queue[0];
+    },
+    produce(...args) {
+      queue.push(args);
+      consumers.length && consumers.shift()(...queue.shift());
+    }
+  };
+}
+
+const speechRecognitionQueue = createProducerConsumer();
```
Comment (corinagum) [inline __tests__/setup/web/mockWebSpeech.js]:
```suggestion
const speechRecognitionBroker = createProducerConsumer();
```

File: __tests__/setup/web/mockWebSpeech.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+const speechSynthesisQueue = createProducerConsumer();
```
Comment (corinagum) [inline __tests__/setup/web/mockWebSpeech.js]:
```suggestion
const speechSynthesisBroker = createProducerConsumer();
```

Comment (corinagum) [reply __tests__/setup/web/mockWebSpeech.js]: update related files

File: __tests__/setup/web/mockWebSpeech.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+class SpeechRecognition extends EventTarget {
+  constructor() {
+    super();
+
+    this.grammars = null;
+    this.lang = 'en-US';
```
Comment (corinagum) [inline __tests__/setup/web/mockWebSpeech.js]: Just verifying understanding: This assumption on locale is only in regards to our tests, right?

File: __tests__/setup/web/mockWebSpeech.js (hunk 1 above)
Comment (compulim) [reply __tests__/setup/web/mockWebSpeech.js]: Fixed.

File: __tests__/setup/web/mockWebSpeech.js (hunk 2 above)
Comment (compulim) [reply __tests__/setup/web/mockWebSpeech.js]: Fixed.

File: __tests__/setup/web/mockWebSpeech.js (hunk 3 above)
Comment (compulim) [reply __tests__/setup/web/mockWebSpeech.js]: Yap. We should add test to make sure Web Chat set `lang` correctly.

Comment (compulim) [reply __tests__/setup/web/mockWebSpeech.js]: Added to our list.

**Pull request merged**